### PR TITLE
docs: Move install section into getting started

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Move install section into getting started
Additional Description:

Further to discussion https://github.com/envoyproxy/envoy/pull/13386#issuecomment-705238469 - this is a breakout PR to move the docs install section into the getting started section

Before we can safely land this we need to add docs redirects (#13436)

Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue] touch #13386 
[Optional Deprecated:]

### CODE REVIEW

**Comment by phlax** [comment]
built docs are here: 

https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/index.html

moved (and hopefully redirected pages):

https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/install/building
https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/install/ref_configs
https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/install/sandboxes/local_docker_build
https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/config_load_check_tool
https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/route_table_check_tool
https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/schema_validator_check_tool
https://389209-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/tools

**Comment by phlax** [comment]
and the redirect from the old install toc page, which has been removed

https://389214-65214191-gh.circle-artifacts.com/0/generated/docs/install/install.html

**Comment by phlax** [comment]
@mattklein123 with these changes in terms of paths  `/install/*` has been moved to `/start/install/*`

i have remove the toc page - previous `/install/install` as it was kinda superflous.

this has left the building section as `/start/install/building` - im minded to move `building` to `/start/building` and add an install page at `/start/install` (see #13461)

**Comment by phlax** [comment]
latest build is here

https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/index.html

the moved/redirected pages are here:

https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/building
https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/ref_configs
https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/sandboxes/local_docker_build
https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/config_load_check_tool
https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/route_table_check_tool
https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/schema_validator_check_tool
https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/tools/tools


the removed/redirected page is here:

https://389246-65214191-gh.circle-artifacts.com/0/generated/docs/install/install

File: docs/root/start/start.rst [hunk 1]
```diff
@@ -12,6 +12,18 @@ These examples use the :ref:`v3 Envoy API <envoy_api_reference>`, but use only t
...
+.. _install:
+
+Building and installation
+-------------------------
+
+.. toctree::
+    :maxdepth: 2
+
+    building
+    install/ref_configs
```
Comment (mattklein123) [inline docs/root/start/start.rst:24]: IMO the ref_configs are somewhat unrelated to building and going through the initial examples. Can you move them somewhere else?

Also, I think start.rst should be a raw TOC page without any text, as it's going to be the entry point to a lot of content. WDYT about moving all of the text in here into smaller independent pages that we can more easily sequence for the right flow?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Very nice. Thanks for working on this. One small comment but otherwise I love this direction!

/wait

File: docs/root/start/start.rst (hunk 1 above)
Comment (phlax) [reply docs/root/start/start.rst:24]: > ref_configs are somewhat unrelated to building

yep, most of the stuff in the `install` folder probs needs to be moved

> I think start.rst should be a raw TOC page without any text

i think probably yes, but atm it is giving the section structure

could we make these changes after landing this?

the redirects, can be chained, so its ok to move and move again, and im  quite up for changing this incrementally

overall tho, defo we want to be able to create a flow through the docs, and the ability to hone in on specific solutions (and SEO to get there etc), so im for moving this around more

Comment (phlax) [reply docs/root/start/start.rst:24]: ill look and see what i can do with `ref_configs` in the meantime

Comment (mattklein123) [reply docs/root/start/start.rst:24]: Yeah sure I'm fine to do this incrementally as long as we agree on the overall direction. LMK if you want to play with this more or merge.

Comment (phlax) [reply docs/root/start/start.rst:24]: lets merge it for now, and ill make some changes when i add the install page

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**